lib/puppet/configurer.rb

```diff
@@ -246,6 +246,12 @@ def run_internal(options)
   private :run_internal
 
   def send_report(report)
+    if Puppet[:show_changes]
```

Comment:
I'm not sure we need an accessor and two different actions, `sumup` and `displaychanges` (should be named `display_changes` too).  I think this would be better as `puts report.display_changes if Puppet[:display_changes]` and let `display_changes` on the report handle summing up and displaying the changes if non-empty.